python/mxnet/contrib/nn/layer.py

```diff
@@ -0,0 +1,355 @@
+# coding: utf-8
+# pylint: disable= arguments-differ
+"""Neural network layers."""
+
+from ... import symbol, ndarray
+from ...symbol import Symbol
+from ...ndarray import NDArray
+from .parameter import Parameter, ParameterDict
+
+
+class Layer(object):
+    """Base class for all neural network layers and models.
+
+    Your models should subclass this class.
+
+    Layers can also contain other Layers, allowing you to nest them in a tree
+    structure. You can assign sublayers as regular attributes::
+        from mxnet import nn
+        class Net(nn.Layer):
+            def __init__(self, prefix=None, params=None):
+                super(Net, self).__init__(prefix=prefix, params=params)
+                self.dense1 = nn.Dense(20, in_units=10, prefix=self.prefix+'dense1_')
+                self.dense2 = nn.Dense(20, in_units=20, prefix=self.prefix+'dense2_')
+
+            def forward(self, x):
+                x = self.dense1(x)
+                return self.dense2(x)
+
+    Sublayers assigned this way will be registered and will have their status changed
+    too when you call .train() etc.
+
+    Parameters
+    ----------
+    prefix : str
+        Prefix acts like a name space. It will be prepended to the name of all Symbols and
+        Parameters created by this layer. Prefix should be unique within one network
+        to prevent name collisions.
+    params : ParameterDict or None
+        Manages Parameters of this Layer and sublayers. You can make two Layers share
+        parameter by passing the same dictionary to them. For example::
+            params = nn.ParameterDict(prefix='dense_')
+            dense1 = nn.Dense(20, in_units=10, prefix='dense1_', params=params)
+            dense2 = nn.Dense(20, in_units=10, prefix='dense2_', params=params)
```

Comment:
What does this example mean? The two separate prefix `dense1_` and `dense2_` are used to name the layer, but their parameter will be named `dense_` XXX?